src/twisted/web/_stan.py

```diff
@@ -288,54 +292,51 @@ def __repr__(self) -> str:
 )
 
 
+@attr.s(hash=False, eq=False, repr=False, auto_attribs=True)
 class CDATA:
     """
     A C{<![CDATA[]]>} block from a template.  Given a separate representation in
     the DOM so that they may be round-tripped through rendering without losing
     information.
-
-    @ivar data: The data between "C{<![CDATA[}" and "C{]]>}".
-    @type data: C{unicode}
     """
 
-    def __init__(self, data):
-        self.data = data
+    data: str
+    """The data between "C{<![CDATA[}" and "C{]]>}"."""
 
     def __repr__(self) -> str:
         return f"CDATA({self.data!r})"
 
 
+@attr.s(hash=False, eq=False, repr=False, auto_attribs=True)
 class Comment:
     """
     A C{<!-- -->} comment from a template.  Given a separate representation in
     the DOM so that they may be round-tripped through rendering without losing
     information.
-
-    @ivar data: The data between "C{<!--}" and "C{-->}".
-    @type data: C{unicode}
     """
 
-    def __init__(self, data):
-        self.data = data
+    data: str
+    """The data between "C{<!--}" and "C{-->}"."""
 
     def __repr__(self) -> str:
         return f"Comment({self.data!r})"
 
 
+@attr.s(hash=False, eq=False, repr=False, auto_attribs=True)
 class CharRef:
     """
     A numeric character reference.  Given a separate representation in the DOM
     so that non-ASCII characters may be output as pure ASCII.
 
-    @ivar ordinal: The ordinal value of the unicode character to which this is
-        object refers.
-    @type ordinal: C{int}
-
     @since: 12.0
     """
 
-    def __init__(self, ordinal):
-        self.ordinal = ordinal
+    ordinal: int
+    """The ordinal value of the unicode character to which this object refers."""
 
     def __repr__(self) -> str:
         return "CharRef(%d)" % (self.ordinal,)
+
+
+if TYPE_CHECKING:
+    from twisted.web.template import Flattenable
```

Comment:
Why not import at the top?

Reply:
I think originally I didn't have the `TYPE_CHECKING` there and then importing at the bottom made sense to reduce the chance of cyclic import problems. I'll move it.